lib/Services/ConvertKit.ts

```diff
@@ -0,0 +1,76 @@
+import * as Http from "@effect/platform/HttpClient";
+import {
+  Config,
+  ConfigSecret,
+  Context,
+  Data,
+  Effect,
+  Layer,
+  flow,
+} from "effect";
+import * as AppSchema from "../Schema";
+import { ParseResult } from "@effect/schema";
+
+export interface ConvertKitConfig {
+  readonly baseUrl: string;
+  readonly apiKey: ConfigSecret.ConfigSecret;
+  readonly formId: string;
+}
+
+export class ConvertKitError extends Data.TaggedError("ConvertKitError")<{
+  readonly reason: "Http";
+  readonly error: unknown;
+}> {}
+
+export interface ConvertKitService {
+  /**
+   * Add new subscriber with given `email`.
+   */
+  readonly addSubscriber: (
+    email: string
+  ) => Effect.Effect<
+    never,
+    ConvertKitError | ParseResult.ParseError,
+    AppSchema.SubscribeResponse
+  >;
+}
+
+export const ConvertKitService = Context.Tag<ConvertKitService>(
+  "@app/ConvertKitService"
+);
+
+const make = ({ baseUrl, apiKey, formId }: ConvertKitConfig) => {
+  const client = Http.client.fetchOk().pipe(
```

Comment:
I find it useful to create a re-usable client first, that adds the common setup for api you are using. Often you would add the api key here too, but the convertkit api is a bit weird in that regard.